File: EdlerCareApi/Models/UserProfiles/UserProfile.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace EdlerCareApi.Models.UserProfiles
{
    [Table("UserProfile", Schema = "public")]
    public class UserProfile : IAuditable
    {
        public Guid Id { get; set; }
        public string FirstName { get; set; }
        public string? LastName { get; set; }
        public string? Email { get; set; }
        public string? Password { get; set; }
        public string? Phone { get; set; }
        public string? Address { get; set; }
        public string? City { get; set; }
        public bool IsDeleted { get; set; } = false;
        public bool IsVerfied { get; set; } = false;
        public string? Username { get; set; }
        public UserProfileRoleType RoleType { get; set; }
        public byte[]? PasswordHash { get; set; }
        public byte[]? PasswordSalt { get; set; }
        public string? RefreshToken { get; set; } = string.Empty;
        [NotMapped]
        public string? Token { get; set; } = string.Empty;
        public DateTime? TokenCreated { get; set; }
        public DateTime? TokenExpires { get; set; }
        public DateTimeOffset CreatedDate { get; set; }
        public DateTimeOffset UpdatedDate { get; set; } = DateTimeOffset.UtcNow;
        public Guid CreatedBy { get; set; }
        public Guid UpdatedBy { get; set; }
    }
}

File: ElderCareClient/src/app/core/models/UserProfile.ts
export class UserProfile {
  Id: string;
  FirstName: string;
  LastName?: string;
  Email?: string;
  Password?: string;
  Phone?: string;
  Address?: string;
  City?: string;
  IsDeleted: boolean;
  IsVerfied: boolean;
  Username: string;
  RoleType: string;
  PasswordHash?: Uint8Array;
  PasswordSalt?: Uint8Array;
  RefreshToken?: string;
  Token?: string;
  TokenCreated?: Date;
  TokenExpires?: Date;
  CreatedDate: Date;
  UpdatedDate: Date;
  CreatedBy: string;
  UpdatedBy: string;

  constructor(
    Id: string,
    FirstName: string,
    LastName: string,
    Email: string,
    Password: string,
    Phone: string,
    Address: string,
    City: string,
    IsDeleted: boolean,
    IsVerfied: boolean,
    Username: string,
    RoleType: string,
    PasswordHash: Uint8Array,
    PasswordSalt: Uint8Array,
    RefreshToken: string,
    Token: string,
    TokenCreated: Date,
    TokenExpires: Date,
    CreatedDate: Date,
    UpdatedDate: Date,
    CreatedBy: string,
    UpdatedBy: string
  ) {
    this.Id = Id;
    this.FirstName = FirstName;
    this.LastName = LastName;
    this.Email = Email;
    this.Password = Password;
    this.Phone = Phone;
    this.Address = Address;
    this.City = City;
    this.IsDeleted = IsDeleted;
    this.IsVerfied = IsVerfied;
    this.Username = Username;
    this.RoleType = RoleType;
    this.PasswordHash = PasswordHash;
    this.PasswordSalt = PasswordSalt;
    this.RefreshToken = RefreshToken;
    this.Token = Token;
    this.TokenCreated = TokenCreated;
    this.TokenExpires = TokenExpires;
    this.CreatedDate = CreatedDate;
    this.UpdatedDate = UpdatedDate;
    this.CreatedBy = CreatedBy;
    this.UpdatedBy = UpdatedBy;
  }
}
